File: pyg-user-web/src/main/webapp/js/service/orderService.js
// 服务层
app.service('orderService', function($http) {
	    	
    // 读取列表数据绑定到表单中
    this.findAll = function () {
        return $http.get('../order/findAll');
    };
    // 分页
    this.findPage = function (page, rows) {
        return $http.get('../order/findPage/' + page + '/' + rows);
    };
    // 查询实体
    this.findOne = function (id) {
        return $http.get('../order/findOne/' + id);
    };
    // 增加
    this.add = function (entity) {
        return $http.post('../order/add', entity);
    };
    // 修改
    this.update = function (entity) {
        return $http.post('../order/update', entity);
    };
    // 删除
    this.del = function (ids) {
        return $http.get('../order/delete/' + ids);
    };
    // 搜索
    this.search = function (page, rows, searchEntity) {
        return $http.post('../order/search/' + page + "/" + rows, searchEntity);
    };

    //订单详情
    this.findOneOrder = function (id) {
        return $http.get("../order/findOneOrder/" + id);
    };

    //查询规格
    this.findSpecForItemId = function (itemId) {
        return $http.get("../order/findSpecForItemId/" + itemId);
    };

    //求合计
    this.sum1 = function(orderDetail){
        var totalValue={totalNum:0, totalMoney:0.00};//合计实体
            for(var j=0;j<orderDetail.orderItemList.length;j++){
                var orderItem=orderDetail.orderItemList[j];//购物车明细
                totalValue.totalNum+=orderItem.num;
                totalValue.totalMoney+= orderItem.totalFee;
            }
        return totalValue;
    };
});

File: pyg-order/pyg-order-interface/src/main/java/com/pyg/order/service/OrderService.java
package com.pyg.order.service;
import java.util.List;

import com.pyg.pojo.TbItem;
import com.pyg.pojo.TbOrder;

import com.pyg.pojo.TbPayLog;
import com.pyg.vo.Order;
import com.pyg.vo.PageResult;

/**
 * Description: 服务层接口
 *
 * @author AK
 * @date 2018/8/9 10:00
 * @since 1.0.0
 */
public interface OrderService {

	/**
     * description: 返回全部列表
     *
     * @return java.util.List<com.pyg.pojo.TbOrder>
     * @author AK
     * @date  2018年08月09日 10:00:00
     */
	List<TbOrder> findAll();
		
	/**
     * description: 分页返回当前用户的订单列表
     *
     * @param page 当前页面
     * @param rows 每页记录数
	 * @param userId 用户id
     * @return com.pyg.vo.PageResult
     * @author AK
     * @date  2018年08月09日 10:00:00
     */
	PageResult findPage(int page, int rows, String userId);
		
	/**
     * description: 增加
     *
     * @param order 实体
     * @return com.pyg.vo.InfoResult
     * @author AK
     * @date  2018年08月09日 10:00:00
     */
	void add(TbOrder order);
		
	/**
     * description: 修改
     *
     * @param order 实体
     * @return com.pyg.vo.InfoResult
     * @author AK
     * @date  2018年08月09日 10:00:00
     */
	void update(TbOrder order);
	
	/**
     * description: 根据id返回实体
     *
     * @param id 实体id
     * @return com.pyg.pojo.TbOrder
     * @author AK
     * @date  2018年08月09日 10:00:00
     */
	TbOrder findOne(Long id);	
	
	/**
     * description: 批量删除
     *
     * @param ids 实体ids
     * @return com.pyg.vo.InfoResult
     * @author AK
     * @date  2018年08月09日 10:00:00
     */
	void delete(Long[] ids);

	/**
     * description: 查询+分页
     *     
     * @param page 当前页
     * @param rows 每页记录数
     * @param order 查询条件
     * @return com.pyg.vo.PageResult
     * @author AK
     * @date  2018年08月09日 10:00:00
     */
	PageResult findPage(Integer page, Integer rows, TbOrder order);

	/**
	 * description: 根据用户查询payLog
	 *
	 * @param userId 用户id
	 * @return com.pyg.pojo.TbPayLog
	 * @author AK
	 * @date  2018年08月29日 09:09:00
	 */
	TbPayLog searchPayLogFromRedis(String userId);

	/**
	 * description: 修改订单状态
	 *
	 * @param out_trade_no 支付订单号
	 * @param transaction_id 微信返回的交易流水号
	 * @return void
	 * @author AK
	 * @date  2018年08月29日 09:15:25
	 */
	void updateOrderStatus(String out_trade_no,String transaction_id);

	/**
	 * description: 根据订单id返回订单包装类对象
	 *
	 * @param id 订单id
	 * @return com.pyg.vo.Order
	 * @author AK
	 * @date  2018年09月06日 21:12:47
	 */
    Order findOrderById(Long id);

	TbItem findSpecForItemId(Long itemId);
}
